Chess/src/GameLogic.cpp

```diff
@@ -0,0 +1,192 @@
+#include "../include/GameLogic.h"
+#include "../include/Chess.h"
+#include "../include/Pawn.h"
+#include "../include/Knight.h"
+#include "../include/Bishop.h"
+#include "../include/Rock.h"
+#include "../include/Queen.h"
+#include "../include/King.h"
+#include <sstream>
+#include <string>
+
+
+using namespace std;
+
+enum behavior { Valid=41, SamePiece, MoveOutOfRange };
+GameLogic::GameLogic(Chess *chess) : chess(chess)
+{
+    PiecesArrays();
+}
+void GameLogic::PiecesArrays()
+{
+    PawnArrays();
+    BishopArrays();
+    KnightArrays();
+    RockArrays();
+    QueenArrays();
+    KingArrays();
+}
+
+int GameLogic::ProcessTheInput(const std::string& m_input)
+{
+    stringstream ss(m_input);
+    string token;
+    vector<pair<int, int>> ParsedInput;
+    while (ss >> token) {
+        pair<int, int> PlaceAsInteger = {0, 0};
+        for (char ch : token) {
+            if (isalpha(ch) && !islower(ch))
+                PlaceAsInteger.first = ((ch - 'A') * 2) + 3;
+            else if (isalpha(ch) && islower(ch))
+                PlaceAsInteger.first = ((ch - 'a') * 2) + 3;
+            else
+            PlaceAsInteger.second = ((ch - '0') * 2) + 1;
+        }
+        ParsedInput.push_back(PlaceAsInteger);
+    }
+
+    int code =  CheckMovement(ParsedInput);
+    codeResponseHandel(code, ParsedInput[0] ,ParsedInput[1], chess->GetPieceAt(ParsedInput[0].first, ParsedInput[0].second));
+    return  code;
+}
+
+int GameLogic::CheckMovement(const std::vector<std::pair<int, int>>& cords)
+{
+    Piece* piece = SearchForPiece(cords[0]);
+    Piece* target = SearchForPiece(cords[1]);
+
+    if (piece)
+    {
+
+        if (chess->ReturnTurn() && piece->GetColor()=='b' || !chess->ReturnTurn() && piece->GetColor() =='w')
+        {
+            return 12;

```

Comment:
Don't use hardcoded values, it will be better to use an `Enum` and give the return values meaningful names. It will make the code more readable.